assets/js/components/settings/SettingsActiveModule/ConfirmDisconnect.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * ConfirmDisconnect component for SettingsActiveModule.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useState, useEffect, useCallback } from '@wordpress/element';
+import { ESCAPE } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_MODULES } from '../../../googlesitekit/modules/datastore/constants';
+import { clearWebStorage } from '../../../util';
+import Dialog from '../../Dialog';
+const { useSelect, useDispatch } = Data;
+
+export default function ConfirmDisconnect( { slug, handleDialog } ) {
+	const [ isDeactivating, setIsDeactivating ] = useState( false );
+
+	const dependentModules = useSelect( ( select ) => select( CORE_MODULES ).getModuleDependantNames( slug ) );
+	const provides = useSelect( ( select ) => select( CORE_MODULES ).getModuleFeatures( slug ) );
+	const module = useSelect( ( select ) => select( CORE_MODULES ).getModule( slug ) );
+	const moduleStoreName = useSelect( ( select ) => select( CORE_MODULES ).getModuleStoreName( slug ) );
+	const adminReauthURL = useSelect( ( select ) => select( moduleStoreName )?.getAdminReauthURL( false ) );
+
+	useEffect( () => {
+		const onKeyPress = ( e ) => {
+			if ( ESCAPE === e.keyCode ) {
+				handleDialog();
+			}
+		};
+
+		global.addEventListener( 'keydown', onKeyPress );
+		return () => {
+			global.removeEventListener( 'keydown', onKeyPress );
+		};
+	}, [] );
+
+	const { deactivateModule } = useDispatch( CORE_MODULES );
+	const handleDisconnect = useCallback( async () => {
+		if ( module.forceActive ) {
+			return;
+		}
+
+		setIsDeactivating( true );
+		const { error } = await deactivateModule( slug );
+		setIsDeactivating( false );
+
+		if ( ! error ) {
+			clearWebStorage();
```

Comment:
I'm not sure we should clear the entire web storage when deactivating a module. Even if we did this before, I think it's too much. Instead I think it would be more appropriate to invalidate/clear the cache for that module, like we do when saving its settings.

@felixarntz